meta-client/src/main/java/org/apache/helix/metaclient/api/MetaClientInterface.java

```diff
@@ -332,9 +349,9 @@ void asyncCreate(final String key, final T data, final EntryMode mode,
   /**
    * Maintains a connection with underlying metadata service based on config params. Connection
    * created by this method will be used to perform CRUD operations on metadata service.
-   * @return True if connection is successfully established.
+   * @Throws MetaClientException when failed to connect.
    */
-  boolean connect();
+  void connect();
```

Comment:
shall we mark exception type at the method?

Reply:
also nit, some behavior clarification on repeated connect() will be good.

Reply:
TFTR. Added a `todo`. Will add the connect/disconnect part in following PR since it needs a bit more error handling and it is orthogonal to this CRUD change.